src/promise.js

```diff
@@ -14,19 +14,29 @@ var UNDEFINED_BINDING = {};
 var ASSERT = require("./assert");
 var util = require("./util");
 
-var getDomain;
+var getContext;
 if (util.isNode) {
-    getDomain = function() {
-        var ret = process.domain;
-        if (ret === undefined) ret = null;
-        return ret;
-    };
+    if (util.nodeSupportsAsyncResource) {
+        var AsyncResource = require("async_hooks").AsyncResource;
+        getContext = function() {
+            return {
+                domain: process.domain,
+                async: new AsyncResource("Promise")
```

Comment:
You should consider calling it `bluebird.Promise` to not confuse it with the builtin `PROMISE`.